src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -29,17 +29,17 @@ abstract class BaseNode implements NodeInterface
     private static array $placeholderUniquePrefixes = [];
     private static array $placeholders = [];
 
-    protected $name;
-    protected $parent;
-    protected $normalizationClosures = [];
-    protected $normalizedTypes = [];
-    protected $finalValidationClosures = [];
-    protected $allowOverwrite = true;
-    protected $required = false;
-    protected $deprecation = [];
-    protected $equivalentValues = [];
-    protected $attributes = [];
-    protected $pathSeparator;
+    protected ?string $name;
```

Comment:
```suggestion
    protected string $name;
```

The constructor makes sure that `$name` is not `null`.

https://github.com/symfony/symfony/blob/f10050724a1d2b5a9d57a30aa730be487d620283/src/Symfony/Component/Config/Definition/BaseNode.php#L51